### PR TITLE
conformed list_files behaviour on failure with pebble's

### LINKED ISSUE #679: harness does not raise the same exception as ops when `list_files` is not found
In normal charm execution, `list_files` will raise `ops.pebble.APIError` while harness will raise `FileNotFoundError`. This mismatch will cause code to break in testing.

### PR DESCRIPTION
**Pull request by PietroPasotti**
Ensured that an APIError is raised instead of FileNotFoundError on list_files('/not/existing/path'); 

Fixes #679

### CODE REVIEW

File: ops/testing.py [hunk 1]
```diff
@@ -1495,7 +1495,14 @@ def push(
...
     def list_files(self, path: str, *, pattern: str = None,
                    itself: bool = False) -> typing.List[pebble.FileInfo]:
-        files = [self._fs.get_path(path)]
+        try:
+            files = [self._fs.get_path(path)]
+        except FileNotFoundError as e:
+            # conform with the real pebble api
+            raise pebble.APIError(
+                body={}, message=str(e),
+                code=404, status='Not Found')
```
Comment (jnsgruk) [inline ops/testing.py:1504]: Nit alert! This wrapping seems a little over-zealous, perhaps...

```suggestion
            raise pebble.APIError(
              body={}, message=str(e),code=404, status='Not Found')
```

I think the above would be more in-keeping with the style elsewhere in OF.

**Review by jnsgruk — APPROVED** [review]
LGTM, thanks @PietroPasotti!

File: ops/testing.py (hunk 1 above)
Comment (PietroPasotti) [reply ops/testing.py]: good point!

**Review by rwcarlsen — APPROVED** [review]
LGTM

**Comment by sed-i** [comment]
Is changing both to [NotADirectoryError](https://docs.python.org/3/library/exceptions.html#NotADirectoryError) in the next major release possible and desired?

File: ops/testing.py [hunk 2]
```diff
@@ -1495,7 +1495,13 @@ def push(
 
     def list_files(self, path: str, *, pattern: str = None,
                    itself: bool = False) -> typing.List[pebble.FileInfo]:
-        files = [self._fs.get_path(path)]
+        try:
+            files = [self._fs.get_path(path)]
+        except FileNotFoundError as e:
+            # conform with the real pebble api
+            raise pebble.APIError(
+                body={}, code=404, status='Not Found', message=str(e))
```
Comment (rwcarlsen) [inline ops/testing.py:1503]: Would it be worth setting the APIError message equal to what we would see from pebble?  From digging through pebble code - it looks like the error/message pebble provides will be: `open /path/that/does/not/exist: no such file or directory`.

**Review by rwcarlsen — CHANGES_REQUESTED** [review]

**Review by benhoyt — APPROVED** [review]

File: ops/testing.py (hunk 2 above)
Comment (benhoyt) [reply ops/testing.py:1503]: Agreed. Not a must as people shouldn't be testing error strings, but doesn't hurt to be consistent either.

Comment (PietroPasotti) [reply ops/testing.py:1503]: It was my intention to keep it consistent but I couldn't verify the exact error wording. Thanks, will do that!

File: ops/testing.py [hunk 3]
```diff
@@ -1495,7 +1495,14 @@ def push(
 
     def list_files(self, path: str, *, pattern: str = None,
                    itself: bool = False) -> typing.List[pebble.FileInfo]:
-        files = [self._fs.get_path(path)]
+        try:
+            files = [self._fs.get_path(path)]
+        except FileNotFoundError as e:
```
Comment (pengale) [inline ops/testing.py:1500]: The linter is upset about the "as e". Once you get rid of that, we should be good to go!

**Review by pengale — CHANGES_REQUESTED** [review]
+1, after the linter error is fixed.

File: ops/testing.py (hunk 3 above)
Comment (PietroPasotti) [reply ops/testing.py:1500]: huh? this passed the last... N times. Sure.

**Comment by PietroPasotti** [comment]
🎉

**Pull request merged**